mkdocs_click/_docs.py

```diff
@@ -100,19 +106,59 @@ def _make_usage(ctx: click.Context) -> Iterator[str]:
     yield ""
 
 
-def _make_options(ctx: click.Context) -> Iterator[str]:
+def _make_options(ctx: click.Context, style: str = "plain") -> Iterator[str]:
     """Create the Markdown lines describing the options for the command."""
-    formatter = ctx.make_formatter()
-    click.Command.format_options(ctx.command, ctx, formatter)
-    # First line is redundant "Options"
-    # Last line is `--help`
-    option_lines = formatter.getvalue().splitlines()[1:-1]
-    if not option_lines:
-        return
-
-    yield "Options:"
-    yield ""
-    yield "```"
-    yield from option_lines
-    yield "```"
-    yield ""
+
+    def backquote(opts: Iterable[str]) -> List[str]:
+        return [f"`{opt}`" for opt in opts]
+
+    def format_possible_value(opt: click.Option) -> str:
+        param_type = opt.type
+
+        if isinstance(param_type, click.Choice):
+            return f"{param_type.name.upper()} ({' &#x7C; '.join(backquote(param_type.choices))})"
+        elif isinstance(param_type, click.DateTime):
+            return f"{param_type.name.upper()} ({' &#x7C; '.join(backquote(param_type.formats))})"
+        elif isinstance(param_type, (click.IntRange, click.FloatRange)):
+            if param_type.min is not None and param_type.max is not None:
+                return f"{param_type.name.upper()} (between `{param_type.min}` and `{param_type.max}`)"
+            elif param_type.min is not None:
+                return f"{param_type.name.upper()} (`{param_type.min}` and above)"
+            else:
+                return f"{param_type.name.upper()} (`{param_type.max}` and below)"
+        else:
+            return param_type.name.upper()
+
+    if style == "plain":
+        formatter = ctx.make_formatter()
+        click.Command.format_options(ctx.command, ctx, formatter)
+        # First line is redundant "Options"
+        # Last line is `--help`
+        option_lines = formatter.getvalue().splitlines()[1:-1]
+        if not option_lines:
+            return
+
+        yield "__Options:__"
+        yield ""
+        yield "```"
+        yield from option_lines
+        yield "```"
+        yield ""
+    elif style == "table":
+        params = [param for param in ctx.command.get_params(ctx) if isinstance(param, click.Option)]
+        if params[0].opts[0] == "--help":
+            return
+
+        yield "__Options:__"
```

Comment:
Should we consider doing things one step at a time, and leaving the formatting to be consistent with the existing one here?

```suggestion
        yield "Options:"
```

Then we can add this along with "bold-style" `Usage` via eg #14.

(Ideally this PR should focus on "Add table style formatting", and _nothing else_. I may be a bit harsh on this front, but separating PRs with a clear purpose each is crucial to maintainability over time, it's a good habit for us to take early on and stick to. :-))